opentelemetry-sdk/src/opentelemetry/sdk/_metrics/export/__init__.py

```diff
@@ -176,3 +178,49 @@ def _shutdown():
         self._daemon_thread.join()
         self._exporter.shutdown()
         return True
+
+
+class PullMetricExporter(MetricExporter):
+    def __init__(self) -> None:
+        self._collect: Callable[
+            ["PullMetricExporter"], None,
+        ] = None
+
+    @final
+    def _set_collect_callback(
+        self,
+        func: Callable[
+            ["PullMetricExporter"], None,
+        ],
+    ) -> None:
+        """This function is internal to the SDK. It should not be called or overriden by users"""
+        self._collect = func
+
+    @final
+    def collect(self) -> None:
+        if self._collect is not None:
+            self._collect()
+
+
+class PullMetricExporterReader(MetricReader):
+    def __init__(
+        self,
+        exporter: PullMetricExporter,
+    ):
+        super().__init__(preferred_temporality=exporter.preferred_temporality)
+        self._exporter = exporter
+        self._exporter._set_collect_callback(self.collect)
+
+    def _receive_metrics(self, metrics: Iterable[Metric]) -> None:
```

Comment:
Where is this invoked?

Reply:
this is invoked by the `MetricReader`'s `collect` method. You can see in line 212 that `collect` is the callback registered with the exporter.